apps/studio/src/features/editing-experience/components/MoveResourceModal/MoveItem.tsx

```diff
@@ -0,0 +1,85 @@
+import { Suspense, useMemo } from "react"
+import { Text } from "@chakra-ui/react"
+import { Button } from "@opengovsg/design-system-react"
+import { QueryErrorResetBoundary } from "@tanstack/react-query"
+import { useAtomValue } from "jotai"
+import { ErrorBoundary } from "react-error-boundary"
+import { BiFile, BiFolder, BiHome } from "react-icons/bi"
+
+import type { RouterOutput } from "~/utils/trpc"
+import { moveResourceAtom } from "../../atoms"
+
+type MoveItemProps = Pick<
+  RouterOutput["resource"]["getChildrenOf"][number],
+  "permalink" | "type"
+> & {
+  onChangeResourceId: () => void
+  resourceId: string
+}
+
+const SuspendableMoveItem = ({
+  permalink,
+  resourceId,
+  type,
+  onChangeResourceId,
+}: MoveItemProps) => {
+  const movedItem = useAtomValue(moveResourceAtom)
+  const icon: JSX.Element = useMemo(() => {
+    switch (type) {
+      case "Folder":
+        return <BiFolder />
+      case "Page": {
+        return permalink === "/" ? <BiHome /> : <BiFile />
+      }
+    }
+  }, [permalink, type])
+
+  return (
+    <Button
+      variant="clear"
+      w="full"
+      justifyContent="flex-start"
+      color="base.content.default"
+      // NOTE: Homepage is special - it's a page but
+      // we also use it to represent site root
+      isDisabled={
+        (type === "Page" && permalink !== "/") ||
+        resourceId === movedItem?.resourceId
```

Comment:
i think we should invert this logic too (in case we add more types, like in my future PR `Collection` and `CollectionPage`. We should only allow movement to only `Folder` types.

But actually I can do it in my PR when the bug surfaces haha

Reply:
updated sir